blosc/CMakeLists.txt

```diff
@@ -93,7 +93,7 @@ set_target_properties(blosc_shared PROPERTIES
         SOVERSION 1  # Change this when an ABI change happens
     )
 if (MSVC)
-	set_target_properties (blosc_shared PROPERTIES COMPILE_FLAGS -DDLL_EXPORT=TRUE)
+	set_target_properties (blosc_shared PROPERTIES COMPILE_FLAGS -DBLOSC_DLL_EXPORT=TRUE)
```

Comment:
As the first place of the problem, `blosc.h` define `DLL_EXPORT`, also used in `pthread.h`.
Thus, even in `TDM-GCC`, It had been compiled as DLL function, which has only static `pthreads`.

* `blosc.c`
  * `#include "blosc.h"`
  ```c
  #ifdef DLL_EXPORT
      #undef DLL_EXPORT
      #define DLL_EXPORT __declspec(dllexport)
  #else
      #undef DLL_EXPORT
      #define DLL_EXPORT
  #endif
  ```
  * `#include <pthread.h>`...now, `DLL_EXPORT` is defined.
  ```c
  // WINPTHREAD_STATIC condition only TDM-GCC (DLL_EXPORT condition only in MinGW)
  #if defined DLL_EXPORT && !defined WINPTHREAD_STATIC
  #ifdef IN_WINPTHREAD
  #define WINPTHREAD_API __declspec(dllexport)
  #else
  #define WINPTHREAD_API __declspec(dllimport) // line A ... would use this line.
  #endif
  #else
  #define WINPTHREAD_API                       // line B ... should use this line.
  #endif
  ```

Before resolved code, call `pthreads` functions with prefix "___ imp_" as a DLL functions due to use `line A` above.
However, It occurs link error because no such symbols, it is always static functions that `pthreads` function of `TDM-GCC`.

As a solution, it is necessary to treat in order to use `line B`.

1. Don't define `DLL_EXPORT`
1. Define `WINPTHREAD_STATIC`

This time resolved in the proposal 1.

It should be noted that in this approach, call `pthreads` as `DLL` function (Normal is probably this way) in `MinGW` with so small thunk that enters jump to `IAT (Import Address Table)`, because use line is changed from `line A` to `line B`.

I selected this solution, because if define `WINPTHREAD_STATIC`, I think that to cause confusion judging from its name in non-static  `MinGW`.